File: facade/src/main/java/org/tdf/sunflower/types/BlockValidateResult.java
package org.tdf.sunflower.types;

import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;
import org.tdf.common.types.Uint256;
import org.tdf.common.util.HexBytes;

import java.util.List;
import java.util.Map;

@Getter
@Setter
public class BlockValidateResult extends ValidateResult {
    private long gas;
    private Uint256 fee;
    private Map<HexBytes, VMResult> results;
    private List<TransactionInfo> infos;

    public BlockValidateResult(boolean success, @NonNull String reason) {
        super(success, reason);
    }

    public static BlockValidateResult fault(String reason) {
        return new BlockValidateResult(false, reason);
    }

    public static BlockValidateResult success() {
        return new BlockValidateResult(true, "");
    }
}

File: facade/src/main/java/org/tdf/sunflower/types/BlockValidateResult.kt
package org.tdf.sunflower.types

import org.tdf.common.types.Uint256
import org.tdf.common.util.HexBytes
import org.tdf.sunflower.facade.TransactionInfo

class BlockValidateResult(
    success: Boolean, reason: String,
    val gas: Long = 0,
    val fee: Uint256 = Uint256.ZERO,
    val results: Map<HexBytes, VMResult> = emptyMap(),
    val infos: List<TransactionInfo> = emptyList()
) : ValidateResult(success, reason) {
    companion object {
        @JvmStatic
        fun fault(reason: String): BlockValidateResult {
            return BlockValidateResult(false, reason)
        }

        @JvmStatic
        fun success(
            gas: Long = 0,
            fee: Uint256 = Uint256.ZERO,
            results: Map<HexBytes, VMResult> = emptyMap(),
            indices: List<TransactionInfo> = emptyList()
        ): BlockValidateResult {
            return BlockValidateResult(true, "", gas, fee, results, indices)
        }
    }
}